invisible_cities/reco/psf_functions_test.py

```diff
@@ -0,0 +1,84 @@
+    add_variable_weighted_mean(hdst, 'Y', 'E', 'Ypeak')
+
+    assert np.allclose(x_mean, hdst.Xpeak.unique())
+    assert np.allclose(y_mean, hdst.Ypeak.unique())
+
+
+def test_add_empty_sensors_and_normalize_q(ICDATADIR):
+    PATH_IN        = os.path.join(ICDATADIR, "exact_Kr_tracks_with_MC.h5")
+    hdst           = load_dst(PATH_IN, 'RECO', 'Events')
+    group          = hdst.groupby(['event'], as_index=False)
+    hdst_processed = group.apply(add_empty_sensors_and_normalize_q    ,
+                                 var      = ['X', 'Y']                ,
+                                 ranges   = [[-50, 50], [-50, 50]]    ,
+                                 database = load_db.DataSiPM('new', 0))
+    hdst_processed.reset_index(inplace=True, drop=True)
+
+    assert np.allclose(hdst_processed.groupby('event').NormQ.sum().values, 1.0)
+    assert hdst_processed.E.sum()     == hdst.E.sum()
+    assert hdst_processed.Q.sum()     == hdst.Q.sum()
+
+
+def test_add_empty_sensors_and_normalize_q_file(ICDATADIR):
+    PATH_IN        = os.path.join(ICDATADIR, "exact_Kr_tracks_with_MC.h5")
+    PATH_TEST      = os.path.join(ICDATADIR, "exact_Kr_tracks_with_MC_psf_empty_sensors.h5")
+    hdst           = load_dst(PATH_IN, 'RECO', 'Events')
+    group          = hdst.groupby(['event'], as_index=False)
+    hdst_processed = group.apply(add_empty_sensors_and_normalize_q    ,
+                                 var      = ['X', 'Y']                ,
+                                 ranges   = [[-50, 50], [-50, 50]]    ,
+                                 database = load_db.DataSiPM('new', 0))
+    hdst_processed.reset_index(inplace=True, drop=True)
+    hdst_psf       = pd.read_hdf(PATH_TEST)
+
+    assert_dataframes_close(hdst_psf, hdst_processed)
+
+
+def test_hdst_psf_processing(ICDATADIR):
+    PATH_IN        = os.path.join(ICDATADIR, "exact_Kr_tracks_with_MC.h5")
+    PATH_TEST      = os.path.join(ICDATADIR, "exact_Kr_tracks_with_MC_psf.h5")
+
+    hdst           = load_dst(PATH_IN, 'RECO', 'Events')
+    hdst_processed = hdst_psf_processing(hdst                                 ,
+                                         ranges   = [[-50, 50], [-50, 50]]    ,
+                                         database = load_db.DataSiPM('new', 0))
+    hdst_psf       = pd.read_hdf(PATH_TEST)
+
+    assert_dataframes_close(hdst_psf, hdst_processed)
+
+
+def test_create_psf(ICDATADIR):
+    PATH_IN        = os.path.join(ICDATADIR, "exact_Kr_tracks_with_MC_psf.h5")
+    PATH_TEST      = os.path.join(ICDATADIR, "test_psf.npz")
+
+    hdst           = pd.read_hdf(PATH_IN)
+    psf            = np.load(PATH_TEST)
+
+    bin_edges = [np.linspace(-50, 50, 101) for i in range(2)]
+    psf_val, entries, binss = create_psf((hdst.RelX, hdst.RelY), hdst.NormQ, bin_edges)
+
+    np.testing.assert_allclose(psf['psf'    ], psf_val)
+    np.testing.assert_allclose(psf['entries'], entries)
+    np.testing.assert_allclose(psf['bins'   ],   binss)
```

Comment:
```suggestion
    assert np.allclose(psf['psf'    ], psf_val)
    assert np.allclose(psf['entries'], entries)
    assert np.allclose(psf['bins'   ],   binss)
```

It's not wrong. It's just to have a more uniform style

Reply:
Actually I've just noticed that other files use `assert_allclose` as well. At least this keeps the style within the file. We should establish a style at some point.